libnmstate/nm/applier.py

```diff
@@ -328,9 +328,8 @@ def prepare_proxy_ifaces_desired_state(ifaces_desired_state):
         port_opts_metadata = iface_desired_state.get(BRPORT_OPTIONS_METADATA)
```

Comment:
Please explain in the commit why it does not add any value. Why was it there in the first place? What did it actually do?

Reply:
What did I miss? I explained it was a `step` that does not seem to be needed.
If it would have given something interested and was worth changing, then I guess it would have merit an explanation of what it currently does (but now it does nothing at all).

Will changing `step` to `abstraction level` help clarify this?

Reply:
What is missing is why you came to the conclusion that it does not add value. Also it says `at the moment`, so will it be needed later? To me it is not clear what the removed code did and why it is not needed anymore, therefore it would be nice to explain it.

Reply:
Sound to me like you are asking for a recursive "why".
Why? Because it does nothing now.
Why? Because the it "translate" a dict to a dict with same key to same value.
why? Because in the past we thought it makes sense to separate between the nmstate state and nm provider state using a translation step.
why? nm api could be independent from nmstate schema.
why? nm could be then easily re-used by other projects.
...
I can continue on for many more.

> Also it says at the moment, so will it be needed later?

Yes, it may come back, I cannot say. I do not see it happening anytime soon.
It just expresses that *now* it has no value, I hoped it is enough reasoning to merit its removal.

Reply:
> Sound to me like you are asking for a recursive "why".

> Why? Because the it "translate" a dict to a dict with same key to same value.
> why? Because in the past we thought it makes sense to separate between the nmstate state and nm provider state using a translation step.
> why? nm api could be independent from nmstate schema.
> why? nm could be then easily re-used by other projects.

Write this in the commit message and the change would be properly motivated and easy to understand.

Reply:
Asking this for an idle code that does nothing is redundant IMO, especially when the base reasoning has been given.
Adding more reasoning per your request.

Reply:
Thank you for explaining this. To me "does not add value" is not that clear but 'it "translate" a dict to a dict with same key to same value.' is.

Reply:
Updated